plugins/inputs/mavlink/README.md

```diff
@@ -0,0 +1,129 @@
+# MavLink Input Plugin
+
+The `mavlink` plugin connects to a MavLink-compatible flight controller such as
+ [ArduPilot](https://ardupilot.org/) or [PX4](https://px4.io/). and translates
+all incoming messages into metrics.
+
+The purpose of this plugin is to allow Telegraf to be used to ingest live
+ flight metrics from unmanned systems (drones, planes, boats, etc.)
+
+Warning: This input plugin potentially generates a large amount of data! Use
+the configuration to limit the set of messages, or use another telegraf plugin
+to filter the output.
+
+## Global configuration options <!-- @/docs/includes/plugin_config.md -->
+
+In addition to the plugin-specific configuration settings, plugins support
+additional global and plugin configuration settings. These settings are used to
+modify metrics, tags, and field or create aliases and configure ordering, etc.
+See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
+
+[CONFIGURATION.md]: ../../../docs/CONFIGURATION.md#plugins
+
+## Configuration
+
+```toml @sample.conf
+# Read metrics from a Mavlink connection to a flight controller.
+[[inputs.mavlink]]
+     # Flight controller URL. Must be a valid Mavlink connection string in one
+     # of the following formats:
+     #
+     # - Serial port:  serial://<device name>:<baud rate> 
+     #            eg: "serial:///dev/ttyACM0:57600"
+     # 
+     # - TCP client:   tcp://<target ip or hostname>:<port>
+     #            eg: "tcp://192.168.1.12:14550"
+     # 
+     # - UDP client:   udp://<target ip or hostname>:<port>
+     #            eg: "udp://192.168.1.12:14550"
+     # 
+     # - UDP server:   udp://:<listen port>
+     #            eg: "udp://:14540"
+     # 
+     # The meaning of each of these modes is documented by
+     # https://mavsdk.mavlink.io/v1.4/en/cpp/guide/connections.html.
+     fcu_url = "udp://:14540"
+
+     # Filter to specific messages. Only the messages in this list will be parsed.
+     # If blank, all messages will be accepted.
+     message_filter = []
+
+     # Mavlink system ID for Telegraf
+     # Only used if the mavlink plugin is sending messages, eg.
+     # when `stream_request_enable` is enabled (see below.)
+     system_id = 254
+
+     # Determines whether the plugin sends requests to stream telemetry,
+     # and if enabled, the requested frequency of telemetry in Hz.
+     # This setting should be disabled if your software controls rates using
+     # REQUEST_DATA_STREAM or MAV_CMD_SET_MESSAGE_INTERVAL
+     # (See https://mavlink.io/en/mavgen_python/howto_requestmessages.html#how-to-request--stream-messages)
+     stream_request_enable = true
+     stream_request_frequency = 4
+```
+
+### Note: Mavlink Dialects
+
+This plugin currently only uses the ArduPilot-specific dialect, which also
+includes messages from the common Mavlink dialect.
+
+See the [Mavlink docs](https://mavlink.io/en/messages/ardupilotmega.html) for
+more info on dialects.
```

Comment:
Is this intended to be a sub-section of **Configuration**?

Reply:
I've just removed this and merged it into the copy in the plugin summary at the top, doesn't need its own section.